### PR TITLE
[DOC] Improve documentation for GLM `second_level` submodule

### PR DESCRIPTION
**Pull request by NicolasGensollen**
This PR proposes to improve the docstrings of public elements of `nilearn.glm.second_level` (and related objects).
It also adds the `HTMLReport` class to the reference documentation as this class was public and exposed to users.

### CODE REVIEW

**Review by bthirion — COMMENTED** [review]
Looks good so far !

File: nilearn/glm/_base.py [hunk 1]
```diff
@@ -83,76 +93,79 @@ def generate_report(
...
+            Cluster size threshold, in :term:`voxels<voxel>`.
             Default=0
 
-        height_control : string or None, optional
-            false positive control meaning of cluster forming
-            threshold: 'fpr' (default) or 'fdr' or 'bonferroni' or None
-            Default='fpr'.
+        height_control : :obj:`str` or None, optional
+            :term:`False positive control<FPR correction>` meaning of cluster
+            forming threshold: 'fpr' (default) or 'fdr' or 'bonferroni' or
```
Comment (ymzayek) [inline nilearn/glm/_base.py:132]:
```suggestion
            forming threshold: 'fpr', 'fdr', 'bonferroni' or
```
Remove redundant default statement

File: nilearn/glm/_base.py [hunk 2]
```diff
@@ -83,76 +93,79 @@ def generate_report(
...
+                - 'l' - sagittal left hemisphere only
+                - 'r' - sagittal right hemisphere only
+                - 'ortho' - three cuts are performed in orthogonal directions
+
+            Default is 'z' if ``plot_type`` is 'slice'; 'ortho' if
+            ``plot_type`` is 'glass'.
 
-        report_dims : Sequence[int, int], optional
+        report_dims : Sequence[ :obj:`int`, :obj:`int` ], optional
             Default is (1600, 800) pixels.
```
Comment (ymzayek) [inline nilearn/glm/_base.py:158]:
```suggestion
```

File: nilearn/glm/_base.py [hunk 3]
```diff
@@ -83,76 +93,79 @@ def generate_report(
...
             Specifies width, height (in pixels) of report window
             within a notebook.
             Only applicable when inserting the report into a Jupyter notebook.
-            Can be set after report creation using report.width, report.height.
+            Can be set after report creation using ``report.width``,
+            ``report.height``.
```
Comment (ymzayek) [inline nilearn/glm/_base.py:163]:
```suggestion
            ``report.height``. Default=(1600, 800).
```

File: nilearn/glm/first_level/design_matrix.py [hunk 4]
```diff
@@ -426,19 +426,20 @@ def make_second_level_design_matrix(subjects_label, confounds=None):
...
 
-    confounds : pandas DataFrame, optional
-        If given, contains at least two columns, 'subject_label' and one
+    confounds : :class:`pandas.DataFrame` or ``None``, optional
+        If given, contains at least two columns, ``subject_label`` and one
         confound. The subjects list determines the rows to extract from
-        confounds thanks to its 'subject_label' column. All subjects must
+        confounds thanks to its ``subject_label`` column. All subjects must
         have confounds specified. There should be only one row per subject.
+        Default=``None``.
```
Comment (ymzayek) [inline nilearn/glm/first_level/design_matrix.py:438]:
```suggestion
        Default=None.
```

**Review by ymzayek — COMMENTED** [review]
@NicolasGensollen thanks for working on this. I made a first pass and mostly everything looks good! Just some minor suggestions.

File: nilearn/reporting/html_report.py [hunk 5]
```diff
@@ -233,13 +233,25 @@ def generate_report(estimator):
 
 class HTMLReport(HTMLDocument):
     """A report written as HTML.
-    Methods such as save_as_html(), open_in_browser()
-    are inherited from HTMLDocument
+
+    Methods such as ``save_as_html``, or ``open_in_browser``
+    are inherited from ``HTMLDocument``.
```
Comment (ymzayek) [inline nilearn/reporting/html_report.py:238]:
```suggestion
    are inherited from :class:`~nilearn.plotting.HTMLDocument`.
```

File: nilearn/plotting/html_document.py [hunk 6]
```diff
@@ -61,17 +60,39 @@ def _check_n_open(self):
...
-        Get the document wrapped in an inline frame.
+        """Get the document wrapped in an inline frame.
 
         For inserting in another HTML page of for display in a Jupyter
         notebook.
 
+        Parameters
+        ----------
+        width : :obj:`int` or ``None``, optional
+            Width of the inline frame. Default=``None``.
```
Comment (ymzayek) [inline nilearn/plotting/html_document.py:86]:
```suggestion
            Width of the inline frame. Default=None.
```

File: nilearn/plotting/html_document.py [hunk 7]
```diff
@@ -61,17 +60,39 @@ def _check_n_open(self):
...
+
+        height : :obj:`int` or ``None``, optional
+            Height of the inline frame. Default=``None``.
```
Comment (ymzayek) [inline nilearn/plotting/html_document.py:89]:
```suggestion
            Height of the inline frame. Default=None.
```

File: nilearn/glm/second_level/second_level.py [hunk 8]
```diff
@@ -278,45 +278,37 @@ def _process_second_level_input_as_firstlevelmodels(second_level_input):
...
         If 0 prints nothing. If 1 prints final computation time.
-        If 2 prints masker computation details. Default=0.
-
-    n_jobs : integer, optional
-        The number of CPUs to use to do the computation. -1 means
-        'all CPUs', -2 'all CPUs but one', and so on.
-        Default=1.
-
-    minimize_memory : boolean, optional
+        If 2 prints masker computation details.
```
[the remark this answers is not in the document]
Comment (ymzayek) [reply nilearn/glm/second_level/second_level.py:309]: Ok!

File: nilearn/reporting/html_report.py [hunk 9]
```diff
@@ -233,13 +233,25 @@ def generate_report(estimator):
...
 
     """
     def __init__(self, head_tpl, body, head_values={}):
-        """The head_tpl is meant for display as a full page, eg writing on
-        disk. The body is used for embedding in an existing page.
+        """Constructor the ``HTMLReport`` class.
+
+        Parameters
+        ----------
+        head_tpl : Template
```
Comment (bthirion) [inline nilearn/reporting/html_report.py:246]: Can the be made more precise ?
Thx ! 
LGTM otherwise.

**Comment by ymzayek** [comment]
@NicolasGensollen thanks for all the changes! One more thing I was wondering concerning this comment above https://github.com/nilearn/nilearn/pull/3030#discussion_r735586892. If I understand correctly, the conclusion is that `second_level_input` (and maybe also `second_level_contrast`) is supposed to be the same for `SecondLevelModel` and `non_parametric_inference`. This is not yet reflected in the docstring of the latter but it should be right?

File: nilearn/reporting/html_report.py (hunk 9 above)
Comment (NicolasGensollen) [reply nilearn/reporting/html_report.py:246]: Indeed, this is not super precise... 😅 
I added a bit more infos in [29e66ec](https://github.com/nilearn/nilearn/pull/3030/commits/29e66ece654aab4206ceaf4d71d4dee93818c7ee)

**Comment by ymzayek** [comment]
Ok thanks for clarifying. Then everything LGTM, I will have a final look and approve once the tests finish running.

**Review by ymzayek — APPROVED** [review]
LGTM, thx!

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Pull request merged**